src/hooks/useGridDimensions.ts

```diff
@@ -0,0 +1,38 @@
+import { useRef, useState, useLayoutEffect } from 'react';
+
+// https://github.com/microsoft/TypeScript/issues/37861
+interface ResizeObserverEntry {
+  contentRect: {
+    width: number;
+    height: number;
+  };
+}
+
+type ResizeObserver = new (callback: (entries: readonly ResizeObserverEntry[]) => void) => {
+  observe: (target: Element) => void;
+  disconnect: () => void;
+};
+
+export function useGridDimensions(): [React.RefObject<HTMLDivElement>, number, number] {
+  const gridRef = useRef<HTMLDivElement>(null);
+  const [gridWidth, setGridWidth] = useState(1);
+  const [gridHeight, setGridHeight] = useState(1);
```

Comment:
I've set to `1` by default just to avoid divisions by zero possibly breaking things.